docs/src/components/main-nav.tsx

```diff
@@ -16,31 +16,31 @@ export interface NavItem {
   label?: string;
 }
 
-export function MainNav(props: { items: NavItem[] }) {
-  const pathname = usePathname();
+function isActive(href: string) {
+  const segment = useSelectedLayoutSegment();
```

Comment:
this doesn't adhere to rules of hooks

Reply:
i have moved it inside the component
```tsx
export function MainNav() {
  const isActive = (href: string) => {
    const segment = useSelectedLayoutSegment();
    if (!segment) return false;
    return href.startsWith(`/${segment}`);
  };
// ...
}
```
is this okay??